### PR TITLE
Restore unit tests

### PR DESCRIPTION
**Pull request by MiaAltieri**
Unit tests for failure scenarios of restore. Successful restore functionality is tested with integration tests

### CODE REVIEW

File: tests/unit/test_mongodb_backups.py [hunk 1]
```diff
@@ -577,3 +577,85 @@ def test_generate_backup_list_output(self, check_output):
...
+
+        self.harness.add_relation(RELATION_NAME, "s3-integrator")
+        self.harness.charm.backups._on_restore_action(action_event)
+
+        action_event.defer.assert_called()
+
+    @patch("charm.subprocess.check_output")
+    @patch("charm.snap.SnapCache")
+    def test_restore_running_backup(self, snap, output):
+        """Verifies backup is fails if another backup is already running."""
```
Comment (marceloneppel) [inline tests/unit/test_mongodb_backups.py:609]:
```suggestion
        """Verifies restore is fails if another backup is already running."""
```

File: tests/unit/test_mongodb_backups.py [hunk 2]
```diff
@@ -577,3 +577,85 @@ def test_generate_backup_list_output(self, check_output):
...
+        action_event = mock.Mock()
+        action_event.params = {"backup-id": "back-me-up"}
+
+        self.harness.charm.backups._on_restore_action(action_event)
+        action_event.fail.assert_called()
+
+    @patch("charm.subprocess.check_output")
+    @patch("charm.snap.SnapCache")
+    def test_restore_syncing(self, snap, output):
+        """Verifies backup is deferred if more time is needed to resync."""
```
Comment (marceloneppel) [inline tests/unit/test_mongodb_backups.py:592]:
```suggestion
        """Verifies restore is deferred if more time is needed to resync."""
```

File: tests/unit/test_mongodb_backups.py [hunk 3]
```diff
@@ -577,3 +577,85 @@ def test_generate_backup_list_output(self, check_output):
         self.assertEqual(failed_backup, "2000-02-14T14:09:43Z  | logical      | finished")
         inprogress_backup = formatted_output[4]
         self.assertEqual(inprogress_backup, "2023-02-14T17:06:38Z  | logical      | in progress")
+
+    def test_restore_without_rel(self):
+        """Verifies no backups are attempted without s3 relation."""
```
Comment (marceloneppel) [inline tests/unit/test_mongodb_backups.py:582]:
```suggestion
        """Verifies no restores are attempted without s3 relation."""
```

File: tests/unit/test_mongodb_backups.py [hunk 4]
```diff
@@ -577,3 +577,85 @@ def test_generate_backup_list_output(self, check_output):
...
+
+        self.harness.add_relation(RELATION_NAME, "s3-integrator")
+        self.harness.charm.backups._on_restore_action(action_event)
+
+        action_event.fail.assert_called()
+
+    @patch("charm.subprocess.check_output")
+    @patch("charm.snap.SnapCache")
+    def test_restore_wrong_cred(self, snap, output):
+        """Verifies backup is fails if the credentials are incorrect."""
```
Comment (marceloneppel) [inline tests/unit/test_mongodb_backups.py:626]:
```suggestion
        """Verifies restore is fails if the credentials are incorrect."""
```

File: tests/unit/test_mongodb_backups.py [hunk 5]
```diff
@@ -577,3 +577,85 @@ def test_generate_backup_list_output(self, check_output):
...
+
+        self.harness.add_relation(RELATION_NAME, "s3-integrator")
+        self.harness.charm.backups._on_restore_action(action_event)
+        action_event.fail.assert_called()
+
+    @patch("charm.subprocess.check_output")
+    @patch("charm.MongoDBBackups._get_pbm_status")
+    @patch("charm.snap.SnapCache")
+    def test_restore_failed(self, snap, pbm_status, output):
+        """Verifies backup is fails if the pbm command failed."""
```
Comment (marceloneppel) [inline tests/unit/test_mongodb_backups.py:645]:
```suggestion
        """Verifies restore is fails if the pbm command failed."""
```

**Review by marceloneppel — APPROVED** [review]
LGTM! My comments are related only to the docstrings.

**Review by delgod — APPROVED** [review]

**Pull request merged**